### PR TITLE
libucontext: new package

### PR DESCRIPTION
**Pull request by cpanato**
- libucontext: new package

Related: https://github.com/chainguard-dev/image-requests/issues/359

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [ ] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams
- [x] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

### CODE REVIEW

File: libucontext.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/kaniini/libucontext
+      tag: v${{package.version}}
+      expected-commit: 4dde3417b4bb4b1b1545bd913be337680b5e28c3
+
+  - runs: |
+      make
+      make install
```
Comment (rawlingsj) [inline libucontext.yaml:25]:
```suggestion
      make DESTDIR="${{targets.destdir}}" install
```
I think we may need this else we won't get the binary in the end apk package.   It looks like `DESTDIR` is supported https://github.com/kaniini/libucontext/blob/be80075e957c4a61a6415c280802fea9001201a2/Makefile#L210C41-L210C48

Comment (cpanato) [reply libucontext.yaml]: done

File: libucontext.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    dependencies:
+      provides:
+        - libucontext-dev=${{package.full-version}}
+
+update:
+  enabled: true
+  github:
+    identifier: kaniini/libucontext
+    strip-prefix: v
+    tag-filter: v
```
Comment (rawlingsj) [inline libucontext.yaml:43]:
```suggestion
    use-tag: true
```
sorry - we do still need use-tags on this package though as https://github.com/kaniini/libucontext/releases only uses tags.  Releases are preferred but only if the upstream project uses github releases.

Comment (rawlingsj) [inline libucontext.yaml:43]:
```suggestion
    tag-filter: v
    use-tag: true
```

sorry - we do still need use-tags on this package though as https://github.com/kaniini/libucontext/releases only uses tags.  Releases are preferred but only if the upstream project uses github releases.

**Review by rawlingsj — APPROVED** [review]

File: libucontext.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  - uses: strip
+
+subpackages:
+  - name: "libucontext-dev"
+    description: "ucontext function implementations (development files)"
+    pipeline:
+      - uses: split/dev
+    dependencies:
+      provides:
+        - libucontext-dev=${{package.full-version}}
```
Comment (rawlingsj) [inline libucontext.yaml:36]: this should be
```
    dependencies:
      runtime:
        - libucontext
```

**Pull request merged**